python/amici/sbml_import.py

```diff
@@ -1321,10 +1337,40 @@ def _add_conservation_for_constant_species(
                 # mark species to delete from stoichiometrix matrix
                 species_solver.pop(ix)
 
+        return species_solver
+
+    def _reduce_stoichiometry(self, species_solver, volume_updates) -> List:
+        """
+        Reduces the stoichiometry with respect to conserved quantities
+
+        :param species_solver:
+            List of species indices which remain later in the ODE solver
+
+        :param volume_updates:
+            List with updates for the stoichiometrix matrix accounting for
+            compartment volumes
+
+        :returns volume_updates_solver:
+            List (according to reduced stoichiometry) with updates for the
+            stoichiometrix matrix accounting for compartment volumes
+        """
+
         # prune out constant species from stoichiometric matrix
         self.stoichiometric_matrix = \
             self.stoichiometric_matrix[species_solver, :]
 
+        # updates of stoichiometry (later dxdotdw in ode_exporter) must be
+        # corrected for conserved quantities:
+        volume_updates_solver = []
+        for update in volume_updates:
```

Comment:
x_index, var1, var2, makes it explicit what we loop over and more evident that we have two loops with the same iterator

Reply:
had that one at some point, needed to enumerate over update (hence changed to update) at some point and left it then...
Agree with you!